kratos/tests/test_sensitivity_builder.py

```diff
@@ -0,0 +1,156 @@
+
+        # nodal_neighbour_map, element_neighbour_map = CalculateEntityNeighbourMaps(self.model_part.Elements)
+        # common_entity_map = CalculateCommonEntityMap(self.model_part.Nodes, element_neighbour_map, nodal_neighbour_map)
+
+        self.__InitializeAssignEntityDerivativesToNodesValues(self.model_part.Elements, 3)
+
+        KratosMultiphysics.SensitivityBuilder.AssignElementDerivativesToNodes(
+            self.model_part,
+            self.model_part.ProcessInfo[KratosMultiphysics.DOMAIN_SIZE],
+            KratosMultiphysics.NORMAL_SHAPE_DERIVATIVE,
+            nodal_neighbour_map,
+            1.0/3.0,
+            KratosMultiphysics.SLIP,
+            True)
+
+        self.__CheckAssignEntityDerivativesToNodesValues(nodal_neighbour_map, common_entity_map)
+
+    def test_AssignConditionDerivativesToNodes(self):
+        nodal_neighbour_map = {60: [56, 61], 56: [60, 52], 52: [56, 47], 47: [52, 45], 45: [47, 29], 18: [8, 29], 8: [18, 3], 3: [8, 1], 46: [48, 30], 48: [46, 51], 29: [45, 18], 61: [60, 59], 9: [19, 4], 19: [9, 30], 30: [46, 19], 57: [59, 51], 59: [57, 61], 1: [4, 3], 4: [1, 9], 51: [48, 57]}
+        common_entity_map = {1: {1: [14, 17], 4: [14], 3: [17]}, 3: {3: [6, 17], 8: [6], 1: [17]}, 4: {4: [14, 19], 1: [14], 9: [19]}, 8: {8: [5, 6], 18: [5], 3: [6]}, 9: {9: [10, 19], 19: [10], 4: [19]}, 18: {18: [5, 18], 8: [5], 29: [18]}, 19: {19: [10, 20], 9: [10], 30: [20]}, 29: {29: [8, 18], 45: [8], 18: [18]}, 30: {30: [11, 20], 46: [11], 19: [20]}, 45: {45: [4, 8], 47: [4], 29: [8]}, 46: {46: [7, 11], 48: [7], 30: [11]}, 47: {47: [3, 4], 52: [3], 45: [4]}, 48: {48: [7, 15], 46: [7], 51: [15]}, 51: {51: [15, 16], 48: [15], 57: [16]}, 52: {52: [2, 3], 56: [2], 47: [3]}, 56: {56: [1, 2], 60: [1], 52: [2]}, 57: {57: [12, 16], 59: [12], 51: [16]}, 59: {59: [12, 13], 57: [12], 61: [13]}, 60: {60: [1, 9], 56: [1], 61: [9]}, 61: {61: [9, 13], 60: [9], 59: [13]}}
+
+        # nodal_neighbour_map, condition_neighbour_map = CalculateEntityNeighbourMaps(self.model_part.Conditions)
+        # common_entity_map = CalculateCommonEntityMap(self.model_part.Nodes, condition_neighbour_map, nodal_neighbour_map)
```

Comment:
this test doesn't do anything

Reply:
I kept them so, if someone changes the mdpa (like I did for `normal_calculation_utils` :/) and need to correct hard-coded `nodal_neighbour_map`, `condition_neighbour_map` and `element_neighbour_map` we just have to uncomment those lines to generate correct maps (Ofc in OMP). I didn't want to use `FindGlobalNodalNeighbours` process (which gives the same maps) directly here since it adds another complexity layer, and increases the execution time of the test. What do you think?

Reply:
And there is no way to find neighbour condition Ids for given node atm in Kratos as well. `GetNeighbourIds` methods are not implemented in `FindElementalNeighboursProcess` and `FindConditionsNeighboursProcess`. :/

Reply:
A comment is added